ably/realtime_channel_spec_test.go

```diff
@@ -926,15 +1014,35 @@ func TestRealtimeChannel_RTL5_Detach(t *testing.T) {
 			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, outMsg.Action)
 		}
 
-		// Get channel state to detaching
-		channel.SetState(chDetaching, nil)
+		// Send attach message
+		in <- &proto.ProtocolMessage{
+			Action:  proto.ActionAttached,
+			Channel: channel.Name,
+		}
+
+		ablytest.Instantly.Recv(t, &change, stateChanges, t.Fatalf)
+		if expected, got := ably.ChannelStateAttached, change.Current; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, change)
+		}
+
+		channel.Detach(cancelledContext)
+
+		// Cause a timeout.
```

Comment:
I'm probably missing something here. What's this timeout for?

Reply:
Since we are calling detach to get into detaching state, the afterCall timeout needs to happen, otherwise test will be stuck there... anyways, it's just to get us to detaching state : )

Reply:
Okay, we don't need timeout there, we are already passing cancelled context